### DISCUSSION TITLE
Where to put config data

### DISCUSSION THREAD

**Discussion by Devasta** [ballsteve/xrust#23] [Planning]
I'm mulling over how to implement the support for DTDs, if we implement a complete spec then I think we need to have external configuration possible, for stuff like max recursion depth on entities, trusted URLs for external DTDs, etc.

Any preferences on what to implement? Can probably hardcode some values for now but I'm sure that we'll want a CLI available at some stage, so whatever we do may be set in stone.

**Comment by ballsteve** [comment]
Yes, it will be best to parameterise these settings.

The usual approach is to have a ParserContext structure. ParserContext::new() would create a context with defaults. The object would then have setter (and getter) methods to override the defaults prior to parsing.
